apis/python/src/tiledbsoma/io.py

```diff
@@ -370,7 +370,6 @@ def from_anndata(
 
     anndata.obs_names_make_unique()
     anndata.var_names_make_unique()
-    anndata = util_ann._decategoricalize(anndata)
```

Comment:
Since we got `anndata = ad.read_h5ad(h5ad_file_name, "r")` with the `"r"` this entire data structure is no longer reconstructible (below). So we split out the decategoricalizations to where they're needed.